daemon/cmd/local_node_sync_test.go

```diff
@@ -127,39 +129,36 @@ func TestLocalNodeSync(t *testing.T) {
 		})
 	)
 
-	require.NoError(t, sync.InitLocalNode(context.TODO(), &local))
+	require.NoError(t, sync.InitLocalNode(t.Context(), &local))
 	require.EqualValues(t, 1, fln.done)
 	require.Equal(t, "foo", local.Name)
 	require.Equal(t, "10.0.0.1", local.GetNodeInternalIPv4().String())
 	require.Equal(t, "fc00::11", local.GetNodeInternalIPv6().String())
 	require.Equal(t, map[string]string{"ex": "label", "foo": "bar"}, local.Labels)
 	require.Equal(t, map[string]string{"ex": "annot", "cilium.io/baz": "qux"}, local.Annotations)
-	require.Equal(t, k8stypes.UID("uid1"), local.UID)
-	require.Equal(t, "provider://foobar", local.ProviderID)
+	require.Equal(t, k8stypes.UID("uid1"), local.Local.UID)
+	require.Equal(t, "provider://foobar", local.Local.ProviderID)
 
 	store := node.NewTestLocalNodeStore(local)
-	updates := stream.ToChannel(context.Background(), store.Observable, stream.WithBufferSize(4))
 
-	sync.SyncLocalNode(context.Background(), store)
+	sync.SyncLocalNode(t.Context(), store)
 	require.EqualValues(t, 5, fln.done)
 
+	updates := stream.ToChannel(t.Context(), store)
 	update := <-updates
-	require.Equal(t, map[string]string{"ex": "label", "foo": "bar"}, update.Labels)
```

Comment:
we still appear to be sending more updates down the channel in the fake above, how come we're no longer asserting on them?

Reply:
As it's now backed by a StateDB table we no longer see every intermediate state (which is good!), so we cannot reliably catch these anymore. We can only assert the final state.

Reply:
Added.